File: java/프로그래머스/level-1/로또의 최고 순위와 최저 순위.java
import java.util.Arrays;
import java.util.List;

class Solution {
    
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = new int[2];
        
        int zeroCnt = 0; // lottos에서 0인 개수
        int minWinCnt = 0; // lottos에서 정답의 개수
        for (int i =0; i < lottos.length; i++){  
            // lottos에서 0인 개수를 증가시킨다.
            if (lottos[i] ==0){
                zeroCnt ++;
            }
            
            // lottos에서 정답의 개수를 찾는다.
            if (contains(win_nums,lottos[i])){
                minWinCnt ++;
            }
        }
        
        
        if (minWinCnt <= 1){
            answer[1]= 6;
        }else{
            answer[1]= 7-minWinCnt;
        }
        int maxWinCnt = minWinCnt+zeroCnt;
        
        if (maxWinCnt<= 1){
              answer[0]= 6;
        }else{
            answer[0]= 7-maxWinCnt;
        }
        return answer;
    }
    
    public static boolean contains(final int[] arr, final int key) {
        return Arrays.stream(arr).anyMatch(i -> i == key);
    }


}

File: python/프로그래머스/level-1/로또의최고순위와최저순위.py
def solution(lottos, win_nums):

    zeroCnt = 0
    matchCnt = 0
    for v in lottos:
        if v != 0 and v in win_nums: # 로또 번호가 정답이랑 일치한 경우
            matchCnt +=1
        elif v == 0:
            zeroCnt +=1

    minV = matchCnt
    if minV ==0:
        minV +=1
    maxV = matchCnt + zeroCnt
    if maxV ==0:
        maxV +=1
    answer = [7-maxV, 7-minV]
    return answer